shell/manifests_test.go

```diff
@@ -0,0 +1,91 @@
+package shell
```

Comment:
Wasn't sure how to test the other parts of this, as it requires a k8s cluster and interactive shell / tty, so I added unit tests for just the manifest generation. Open to suggestions about how to test this better, maybe adding E2E in a followup PR.